tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/OpenTelemetry/ResourceAttributeProcessorHelper.cs

```diff
@@ -0,0 +1,96 @@
+// <copyright file="ResourceAttributeProcessorHelper.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System;
+using System.Reflection.Emit;
+using Datadog.Trace.Activity.DuckTypes;
+using Datadog.Trace.Activity.Handlers;
+using Datadog.Trace.DuckTyping;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.OpenTelemetry
+{
+    internal static class ResourceAttributeProcessorHelper
+    {
+        private static Func<object, object>? _getResourceDelegate;
+
+        static ResourceAttributeProcessorHelper()
+        {
+            _getResourceDelegate = CreateGetResourceDelegate();
+        }
+
+        public static void OnStart(object processor, object activityData)
+        {
+            if (_getResourceDelegate is null
+                || !activityData.TryDuckCast<IActivity>(out var activity)
+                || !processor.TryDuckCast<BaseProcessorStruct>(out var baseProcessor))
+            {
+                return;
+            }
+
+            if (DefaultActivityHandler.ActivityMappingById.TryGetValue(activity.Id, out var activityMapping))
```

Comment:
IIRC, when @tonyredondo originally wrote the activity integration, the assumption was that we may have _other_ activity handlers too, and the default handler is just the fallback. I'm not sure we have any at the moment, but if there's a possibility we might, we might want to rethink the simplistic "expose `ActivityMappingById` publicly" approach 🤔

Reply:
I see what you're saying...would it be good enough for now to move the dictionary and the new `ActivityMapping` struct to the `ActivityListenerHandler` static class? This class is sort of the entrypoint for all of our Activity processing

Reply:
I think that probably makes sense, and implies that _all_ handlers should register the activity by ID by default? Not an issue currently, given our only other handler is an "ignore" handler that we specifically ignore, but could potentially be important later 🤷 

On second thoughts, YAGNI? 😅 I'm happy either way tbh, so feel free to leave it!